### PR TITLE
Remove packaged data in favour of using CRiSpData repository

### PR DESCRIPTION
**Pull request by fnattino**
We have created a [dedicated repository](https://github.com/CityRiverSpaces/CRiSpData) with some datasets to be used by all CRiSp-related packages. 

This PR removes the local datasets and use data from that repository instead.

### CODE REVIEW

**Comment by fnattino** [comment]
The dataset included in the CRiSpData repository turned be to be slightly different from what previously present here. In particular, it will be slightly larger and include few edges that forms the following stroke:  

Screenshot 2025-03-31 at 11 41 27

However, the test on real data would fail, not because of issues with `rcoins` but rather because `sf::st_contains`, which we use to identify the edges included in each stroke, does miss the forming edges. I had to adapt and generalize the test, now it should be more robust to this type of edge cases.

File: tests/testthat/test-stroke.R [hunk 1]
```diff
@@ -197,14 +197,17 @@ test_that("a ring is recognized when from_edge is specified", {
...
 
   # find out which of the initial edges are contained in each of the strokes
-  # NOTE: edges included in self-intersecting strokes can be missed by the
-  # following command, if the test fails double check the input!
+  # NOTE: edges that form self-crossing strokes are missed by the "contains"
+  # relationship. However, they are catched by the "overlaps" relationship,
+  # which is why we run both predicate and merge results
   contains <- sf::st_contains(strokes, edges)
+  overlaps <- sf::st_overlaps(strokes, edges)
+  contains_and_overlaps <- mapply(c, contains, overlaps)
```
Comment (fnattino) [inline tests/testthat/test-stroke.R:210]: Changes here due to: https://github.com/CityRiverSpaces/rcoins/pull/49#issuecomment-2765715837

File: README.md [hunk 2]
```diff
@@ -52,8 +52,8 @@ Rscript -e 'devtools::install()'
 Given the street network of the city of Bucharest (data source: [OpenStreetMap][osm])
```
Comment (SarahAlidoost) [inline README.md:52]: Should we add a note that `CRiSpData` must be installed: `Rscript -e 'devtools::install_github("CityRiverSpaces/CRiSpData")'`? or the dev installation code block, above, should include `dependencies=TRUE`

File: README.md [hunk 3]
```diff
@@ -73,6 +73,7 @@ plot(sf::st_geometry(streets),
 Determine continuous lines in the network as:
 
 ``` r
+library(rcoins)
 continuous_streets <- stroke(streets)
 ```
```
Comment (SarahAlidoost) [inline README.md:79]: Since `CRiSpData` is cropped, the figure in the readme `man/figures/continuous-streets.png` should be updated too, but the difference is minor.

File: .github/workflows/compare_momepy.yaml [hunk 4]
```diff
@@ -27,6 +31,9 @@ jobs:
 
       - name: Compare rcoins output to momepy output
         run: |
-          Rscript -e 'devtools::install()'
+          # install packaged data
+          Rscript -e 'devtools::install_github("CityRiverSpaces/CRiSpData")'
+          # install the current package
+          Rscript -e 'devtools::install(dependencies = FALSE)'
```
Comment (SarahAlidoost) [inline .github/workflows/compare_momepy.yaml:37]: why not install the dependencies? CRiSpData is already one of the `Suggests` dependencies, right? for example, we could do `devtools::install(dependencies = "Suggests", quiet = TRUE, upgrade = "always")`

**Review by SarahAlidoost — APPROVED** [review]
@fnattino thanks. The changes look good, I was able to run the test and the example in readme. Just some minor comments.

File: .github/workflows/compare_momepy.yaml (hunk 4 above)
Comment (fnattino) [reply .github/workflows/compare_momepy.yaml]: Worked! :)
